pkg/definition/definition.go

```diff
@@ -331,10 +333,36 @@ func (def *Definition) FromCUE(val *cue.Value, templateString string) error {
 	if err := unstructured.SetNestedField(spec, templateString, DefinitionTemplateKeys[1:]...); err != nil {
 		return err
 	}
+	if err = validateSpec(spec, def.GetType()); err != nil {
+		return fmt.Errorf("invalid definition spec: %w", err)
+	}
 	def.Object["spec"] = spec
 	return nil
 }
 
+func validateSpec(spec map[string]interface{}, t string) error {
+	bs, err := json.Marshal(spec)
+	if err != nil {
+		return err
+	}
+	var tpl interface{}
+	switch t {
+	case "component":
+		tpl = &v1beta1.ComponentDefinitionSpec{}
+	case "trait":
+		tpl = &v1beta1.TraitDefinitionSpec{}
+	case "policy":
+		tpl = &v1beta1.PolicyDefinitionSpec{}
+	case "workflow-step":
+		tpl = &v1beta1.WorkflowStepDefinitionSpec{}
+	default:
```

Comment:
Should we add a warning here?

Reply:
For default case, we can ignore the validation of spec for now. I think by far we only have the "scope" case and "workload" case in default.